--- /usr/ports/security/krb5/Makefile	2015-06-06 20:27:21.000000000 +0000
+++ Makefile.krb5	2015-07-10 14:41:26.000000000 +0000
@@ -37,12 +37,13 @@
 CPE_VERSION=		5-${PORTVERSION}
 CPE_PRODUCT=		kerberos
 
-OPTIONS_DEFINE=		KRB5_PDF KRB5_HTML DNS_FOR_REALM LDAP READLINE
+OPTIONS_DEFINE=		KRB5_PDF KRB5_HTML DNS_FOR_REALM LDAP LDAP_SASL READLINE
 OPTIONS_DEFAULT=	KRB5_PDF KRB5_HTML
 KRB5_PDF_DESC=		Install krb5 PDF documentation
 KRB5_HTML_DESC=		Install krb5 HTML documentation
 DNS_FOR_REALM_DESC=	Enable DNS lookups for Kerberos realm names
 LDAP=			Enable LDAP support
+LDAP_SASL_DESC=     Link against LDAP with SASL support instead
 
 .if defined(KRB5_HOME)
 PREFIX=			${KRB5_HOME}
@@ -77,6 +78,9 @@
 .endif
 
 .if ${PORT_OPTIONS:MLDAP}
+.if ${PORT_OPTIONS:MLDAP_SASL}
+WANT_OPENLDAP_SASL= yes
+.endif
 USE_OPENLDAP=		yes
 CONFIGURE_ARGS+=	--with-ldap
 PLIST_SUB+=		LDAP=""
